### PR TITLE
Update optimizer overview

### PR DESCRIPTION
**Pull request by som-snytt**
Use the simplified options, but reference the old blog post.

Trim a few ambitious promises to explain more.

### CODE REVIEW

**Comment by som-snytt** [comment]
It would be useful to show a `build.sbt` snippet for "enable under CI".

Or, mention `-opt:none` and how to enable during dev workflow.

**Comment by som-snytt** [comment]
This falls somewhere between "no fair!" and "that's life!"
```
Checking 1501 external links...
Ran on 1188 files!
- ./_site/overviews/scaladoc/contribute.html

  *  External link https://dotty.epfl.ch/docs/usage/dottydoc.html failed: 404 No error

- ./_site/scala3/reference/other-new-features/explicit-nulls.html
htmlproofer 3.15.3 | Error:  HTML-Proofer found 3 failures!
  *  External link https://dotty.epfl.ch/docs/internals/explicit-nulls.html failed: 404 No error
- ./_site/scala3/scaladoc.html
  *  External link https://dotty.epfl.ch/docs/usage/scaladoc/blog.html failed: 404 No error
Error: Process completed with exit code 1.
```

File: _overviews/compiler-options/optimizer.md [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+- Enable method-local optimizations with `-opt:local`. This option is safe for binary compatibility, but typically doesn't improve performance on its own.
+- Enable inlining in addition to method-local optimizations with `-opt:inline:[PATTERN]`.
+  - Don't inline from your dependencies when publishing a library, it breaks binary compatibility. Use `-opt:inline:my.package.**` to only inline from packages within your library.
+  - When compiling an application with global inlining (`-opt:inline:**`), ensure that the run-time classpath is **exactly the same** as the compile-time classpath.
+- The `@inline` annotation only has an effect if the inliner is enabled. It tells the inliner to always try to inline the annotated method or callsite.
+- Without the `@inline` annotation, the inliner generally inlines higher-order methods and forwarder methods. The main goal is to eliminate megamorphic callsites due to functions passed as argument, and to eliminate value boxing. Other optimizations are delegated to the JVM.
+
+Read more to learn more.
```
Comment (som-snytt) [inline _overviews/compiler-options/optimizer.md:23]: Consider producing a t-shirt with this slogan, plus the thinking meme with LeVar Burton, if he's willing to donate his likeness.

File: _data/overviews.yml [hunk 2]
```diff
@@ -259,6 +259,11 @@
       icon: cog
       url: "compiler-options/errors.html"
       description: "A new engine for more user-friendly error messages, printing chains of dependent implicits and colored found/required type diffs."
+    - title: Optimizer
+      by: Lukas Rytz and Andrew Marki
+      icon: cog
+      url: "compiler-options/optimizer.html"
+      description: "The compiler can perform various optimizations as specified by compiler options."
```
Comment (SethTisue) [inline _data/overviews.yml:266]: s/as specified by compiler options/in a configurable manner/

File: _overviews/compiler-options/optimizer.md [hunk 3]
```diff
@@ -0,0 +1,223 @@
+---
+layout: singlepage-overview
+title: Optimizer
+---
+
+**[Lukas Rytz](https://github.com/lrytz) (November 7, 2018)**
```
Comment (SethTisue) [inline _overviews/compiler-options/optimizer.md:6]: s/November 7, 2018/2018/

File: _overviews/compiler-options/optimizer.md [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+
+**[Andrew Marki](https://github.com/som-snytt) (February 29, 2022)**
```
Comment (SethTisue) [inline _overviews/compiler-options/optimizer.md:8]: s/February 29, 2022/2022/

and I'll have you know that I SEE WHAT YOU DID THERE, THERE IS NO FEBRUARY 29 2022 AND WE FROWN ON THESE SHENANIGANS

**Review by SethTisue — CHANGES_REQUESTED** [review]
I've requested a few trivial edits.

Maybe the compiler-options page could link to this new page?

**Comment by SethTisue** [comment]
@lrytz I suggested to Som that after this is merged, I could take care of adding a notice to the top of your blog post directing people here instead.

File: _overviews/compiler-options/optimizer.md (hunk 4 above)
Comment (som-snytt) [reply _overviews/compiler-options/optimizer.md:8]: I literally can't get anything by you. I mean, a doc PR during the long tail of pandemic when there's no one available to do reviews, I thought I could sneak in one chuckle.

File: _overviews/compiler-options/optimizer.md (hunk 3 above)
Comment (som-snytt) [reply _overviews/compiler-options/optimizer.md:6]: OK, but it was fun knowing what Lukas was doing as US election results rolled in. That would be midterm results in the Trump admin. Something about the drift in the political landscape made Lukas go, I'd better right the ship of state with a blog about the optimizer. And that's why we are where we are today, modulo Ukraine.

File: _data/overviews.yml (hunk 2 above)
Comment (som-snytt) [reply _data/overviews.yml:266]: > The compiler is pleased to perform such optimizations as the user may wish to configure by way of compiler settings or command line options.

File: _overviews/compiler-options/optimizer.md (hunk 4 above)
Comment (SethTisue) [reply _overviews/compiler-options/optimizer.md:8]: THERE SHALL BE NO HUMOR IN SCALA

**Review by lrytz — APPROVED** [review]
👍 thank you!

File: _data/overviews.yml (hunk 2 above)
Comment (som-snytt) [reply _data/overviews.yml:266]: I deleted the clause as extraneous, though my previous suggestion is pretty good.

**Comment by som-snytt** [comment]
I caved on Seth's points because I just read in the Sylvia Plath bio that Ted Hughes removed three poems from his book because Marianne Moore asked him to.

**Pull request merged**